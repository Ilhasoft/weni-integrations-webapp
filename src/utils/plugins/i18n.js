import Vue from 'vue';
import VueI18n from 'vue-i18n';

Vue.use(VueI18n);

import en from '../../locales/en.json';
import pt_br from '../../locales/pt_br.json';
import es_es from '../../locales/es_es.json';

const languages = {
  'en-us': en,
  'pt-br': pt_br,
  es: es_es,
};

const messages = Object.assign(languages);

const i18n = new VueI18n({
  locale: 'en-us',
  fallbackLocale: 'en-us',
  messages,
});

export default i18n;
