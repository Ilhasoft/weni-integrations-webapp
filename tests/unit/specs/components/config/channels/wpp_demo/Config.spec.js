jest.mock('@/api/appType', () => {
  return {
    updateAppConfig: jest.fn(),
  };
});

// import { unnnicCallAlert as mockUnnnicCallAlert } from '@weni/unnnic-system';

jest.mock('@weni/unnnic-system', () => ({
  ...jest.requireActual('@weni/unnnic-system'),
  unnnicCallAlert: jest.fn(),
}));

import Vuex from 'vuex';
import { shallowMount, createLocalVue } from '@vue/test-utils';
import wppDemoConfig from '@/components/config/channels/wpp_demo/Config.vue';
import { singleApp } from '../../../../../../__mocks__/appMock';
import i18n from '@/utils/plugins/i18n';

const localVue = createLocalVue();
localVue.use(Vuex);

describe('Config.vue', () => {
  let wrapper;
  let actions;
  let store;

  beforeEach(() => {
    actions = {
      updateAppConfig: jest.fn(),
    };

    store = new Vuex.Store({
      actions,
    });

    wrapper = shallowMount(wppDemoConfig, {
      localVue,
      store,
      i18n,
      propsData: {
        app: singleApp,
      },
      mocks: {
        $t: () => 'some specific text',
      },
      stubs: {
        UnnnicInput: true,
        UnnnicButton: true,
        UnnnicIconSvg: true,
        UnnnicModal: true,
      },
    });
  });

  afterEach(() => {
    jest.clearAllMocks();
  });

  it('should be rendered properly', () => {
    expect(wrapper).toMatchSnapshot();
  });

  describe('closePopUp()', () => {
    it('should set showModal oposite', () => {
      wrapper.setData({ showModal: true });
      expect(wrapper.vm.showModal).toBeTruthy();
      wrapper.vm.closePopUp();
      expect(wrapper.vm.showModal).toBeFalsy();
    });

    it('should call parent closePopUp()', () => {
      expect(wrapper.emitted('closePopUp')).toBeFalsy();
      wrapper.vm.closePopUp();
      expect(wrapper.emitted('closePopUp')).toBeTruthy();
    });
  });

  describe('openWppLink()', () => {
    it('should call closePopUp()', () => {
      window.open = jest.fn();
      const spy = spyOn(wrapper.vm, 'closePopUp');
      expect(spy).not.toHaveBeenCalled();
      wrapper.vm.openWppLink();
      expect(spy).toHaveBeenCalledTimes(1);
    });
  });
});
