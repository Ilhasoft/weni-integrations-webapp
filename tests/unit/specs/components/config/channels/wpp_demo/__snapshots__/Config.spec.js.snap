// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Config.vue should be rendered properly 1`] = `
<unnnicmodal-stub text="some specific text" scheme="feedback-green" modal-icon="check-circle-1-1" class="wpp-demo-modal"><span>some specific text</span>
  <unnnicbutton-stub type="terciary">some specific text</unnnicbutton-stub>
  <unnnicbutton-stub type="primary">
    some specific text
  </unnnicbutton-stub>
</unnnicmodal-stub>
`;
